src/lib/chips/chip-list.ts

```diff
@@ -15,13 +15,27 @@ import {
   QueryList,
   ViewEncapsulation,
   OnDestroy,
+  Optional,
+  ElementRef,
+  Renderer2,
 } from '@angular/core';
 
 import {MdChip} from './chip';
 import {FocusKeyManager} from '../core/a11y/focus-key-manager';
-import {SPACE, LEFT_ARROW, RIGHT_ARROW} from '../core/keyboard/keycodes';
+import {BACKSPACE, DELETE, SPACE, DOWN_ARROW, LEFT_ARROW, RIGHT_ARROW, UP_ARROW} from '../core/keyboard/keycodes';
+import {coerceBooleanProperty, Directionality} from '@angular/cdk';
 import {Subscription} from 'rxjs/Subscription';
-import {coerceBooleanProperty} from '@angular/cdk';
+
+/** Utility to check if an input element has no value. */
+function _isInputEmpty(element: HTMLElement): boolean {
```

Comment:
I would move this to the bottom of the file (we typically put module-level functions at the bottom). Also can remove the leading `_` since it's scoped to this module.